### DISCUSSION TITLE
Desktop clients - Windows, MacOS and Linux

### DISCUSSION THREAD

**Discussion by trymeouteh** [ImranR98/Obtainium#1760] [Ideas]
Please create a Obtainium client for the desktop to download apps on Windows, MacOS and Linux.

Would be nice to be able to download these types of packages on each operating system.

- Windows
   - EXEs/MSIs
   - Portable Apps
- MacOS
   - DMG
- Linux
   - Portable (Bare Metal)
   - AppImage

Glad the app is written in Flutter to make such as thing easier, not having to build another app from scratch. However a CLI version on desktop would be nice as.

**Comment by ImranR98** [comment]
I think this is out of scope for Obtainium as far as I'm concerned. I would accept a PR if it was well thought out and didn't add too much additional OS-specific code that needs to be maintained (but I'm not sure that's doable).

**Reply by 2gn** [reply]
An iOS build would be nice, since installing apps from third-party app store is now possible (in EU and Japan).

**Comment by playagain96** [comment]
It's a really cool idea, cause a lot of desktop opensource apps don't update themselves

**Comment by Nichtraucher** [comment]
The [Komi Store](https://komistore.app/) has risen in popularity recently. It has clients for those three OS' plus Android.

**Comment by tsyche** [comment]
ok so it's not a GUI client, but here's a CLI brew companion for the unmanaged case I recently shipped (e.g. no formula, or abandoned/lagging tap).

Installs from GH releases via a config list, if the app ships one. Binaries, .app bundles, AppImages. macOS + Linux/WSL; Windows is planned.
https://github.com/tsyche/untapped

**Comment by 2gn** [comment]
I think you should use package managers for their own platforms. For Windows there's winget, for macos homebrew. On linux you could use [eget](https://github.com/zyedidia/eget).

**Reply by tsyche** [reply]
Agreed.  I just made untapped for apps with dead formulas, unmaintained taps, projects that only ship release assets, etc. On Android that layer has Obtainium; on the desktop it was curl/tar and a note I eventually will lose. 

The goal when i started was obtainium's model but simple, for your terminal:  one conf for the stragglers, one command installs and upgrades.
